src/wp-includes/block-bindings/entity.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Entity source for the block bindings.
+ *
+ * @since 6.9.0
+ * @package WordPress
+ * @subpackage Block Bindings
+ */
+
+/**
+ * Gets value for Entity source.
+ *
+ * @since 6.9.0
+ * @access private
+ *
+ * @param array    $source_args    Array containing source arguments used to look up the override value.
+ *                                 Example: array( "key" => "url" ).
+ * @param WP_Block $block_instance The block instance.
+ * @return mixed The value computed for the source.
+ */
+function _block_bindings_entity_get_value( array $source_args, $block_instance ) {
+	// Get the key from source args - no key means invalid binding
+	if ( empty( $source_args['key'] ) ) {
+		return null;
+	}
+
+	$key = $source_args['key'];
+
+	// For now, only support 'url' key
+	if ( 'url' !== $key ) {
+		return null;
+	}
+
+	// Read entity data from block attributes
+	$entity_id = $block_instance->attributes['id'] ?? null;
+	$type      = $block_instance->attributes['type'] ?? '';
+	$kind      = $block_instance->attributes['kind'] ?? '';
+
+	if ( empty( $entity_id ) ) {
+		return null;
+	}
+
+	try {
+		// Handle post types
+		if ( 'post-type' === $kind ) {
+			$post = get_post( $entity_id );
+			if ( ! $post ) {
+				return null;
+			}
+
+			$permalink = get_permalink( $entity_id );
+			if ( is_wp_error( $permalink ) ) {
+				return null;
+			}
```

Comment:
```suggestion
			if ( is_wp_error( $permalink ) ) {
				return null;
			}
```

The [get_permalink()](https://github.com/WordPress/wordpress-develop/blob/6.8.3/src/wp-includes/link-template.php#L168) return either string or bool false why we check `is_wp_error` here?

Reply:
That's just an error on my part which we need to fix.